src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +190,282 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
+
+	if (! (*oper_db_client)->del((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to delete LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR keys {%u, %s, %s, %s} from oper DB",
+		terminator_id, tenant_id, account_id, vrf_name);
+		return false;
+	}
+
+	return true;
+}
+
+bool oper_db_writer_set_terminator_tunnel_oper_data(const struct terminator_tunnel_oper_data *oper_data, bool write_uptime) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+	const char* keys[] = {oper_data->vrf_name, oper_data->device_id};
+	enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES oper_db_status_value, oper_db_ike_version;
+
+	if(! tunnel_status_to_oper_db(oper_data->status, oper_db_status_value)) {
+		return false;
+	}
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA");
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VRF_NAME, oper_data->vrf_name)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VRF_NAME to %s",
+					oper_data->vrf_name);
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_ID, oper_data->device_id)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_ID to %s",
+					oper_data->device_id);
+		return false;
+	}
+
+	if (! (*oper_db_client)->set((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to write vrf[%s] device-id[%s] tunnel oper data to oper DB", oper_data->vrf_name, oper_data->device_id);
+		return false;
+	}
+
+	db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA");
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_STATUS, oper_db_status_value)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_STATUS to %d",
+					oper_data->status);
+		return false;
+	}
+
+	if (! db_obj->u8_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TERMINATOR_ID, oper_data->term_id)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TERMINATOR_ID to %hu",
+					oper_data->term_id);
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACTIVENESS_STATUS,	oper_data->activeness_status ? LIBOPER_DB_CLIENT_ACTIVENESS_STATUS_ACTIVE :
```

Comment:
As this line is getting pretty long, it will be clearer to have a separate `activeness_status_to_oper_db()` function, as there is for any other field that requires converion.